### ISSUE TITLE
Teleport does not set terminal size correctly on connection

### ISSUE THREAD

**Issue by aashley** [gravitational/teleport#1692]
**What happened**: Terminal size is not set correctly and text wrapping does not work correctly. Fix for #1087 does not work.

**What you expected to happen**: Text to wrap correctly as per every other terminal app.

**How to reproduce it (as minimally and precisely as possible)**:
```
adama@swiftdesk6:~$ tsh version
Teleport v2.4.1 git:v2.4.1-0-gf90cc7c
adama@swiftdesk6:~$ tput cols
90
adama@swiftdesk6:~$ tput lines
57
adama@swiftdesk6:~$ tsh --proxy=teleport.local --cluster=od003 ssh root@od003
root@od003:~# tsh version
Teleport v2.4.1 git:v2.4.1-0-gf90cc7c
root@od003:~# tput cols
80
root@od003:~# tput lines
24
root@od003:~# 
```

This was run in gnome-terminal and mate-terminal as per my original comments on #1087 with the same failure results

**Environment**:
- Teleport version (use `teleport version`):  Teleport v2.4.1 git:v2.4.1-0-gf90cc7c
- Tsh version (use `tsh version`):  Teleport v2.4.1 git:v2.4.1-0-gf90cc7c
- OS (e.g. from /etc/os-release): Ubuntu 16.04.3 LTS

**Browser environment**

- Browser Version (for UI-related issues):
- Install tools:
- Others:

**Relevant Debug Logs If Applicable**

- tsh --debug
- teleport --debug

**Comment by aashley** [comment]
Screenshot of the terminal performing the above tests https://i.imgur.com/mgdcd2Y.png

**Comment by aashley** [comment]
And performing the same test in the same terminal via SSH to the same host.

```
adama@swiftdesk6:~$ ssh -p 23003 root@10.251.8.60 
swift@10.251.8.60's password: 
Welcome to Ubuntu 16.04.3 LTS (GNU/Linux 4.13.0-32-generic x86_64)
Last login: Fri Feb 16 09:36:41 2018 from ::1
root@od003:~# tput cols
90
root@od003:~# tput lines
57
root@od003:~# 
```

**Comment by klizhentas** [comment]
@russjones is this the same issue you fixed by https://github.com/gravitational/teleport/pull/1567 or a different one?

**Comment by aashley** [comment]
I think the fix he did in #1567 fixed it for mac terminals but it didn't make a difference for the others mentioned in #1087. Certainly the behaviour of tsh before and after #1567 has not changed on linux builds within any terminal software I've tried.

**Comment by klizhentas** [comment]
@russjones thoughts?

**Comment by kontsevoy** [comment]
@aashley is that gnome-terminal you're using?

**Comment by russjones** [comment]
@aashley I was able to reproduce this using Ubuntu 16.04 and GNOME Terminal 3.18.3. I'll investigate.

**Comment by russjones** [comment]
I did some more investigation into this, this only seems to be a problem when using the recording proxy and connecting to a Teleport node. Otherwise terminals are re-sized as expected.

| | Teleport Node | OpenSSH |
|-|-|-|
| Regular Teleport | ✖ | ✔ |
| Recording Proxy Teleport | ✔ | ✔ |

@aashley What configuration are you running into this problem?

**Comment by aashley** [comment]
We're using Teleport Node with recording proxy.

Our setup is lots of remote deployed machines on other networks behind NAT firewalls so we use a cloud master with each site being a trusted cluster. Using the recording proxy to maintain logs of the sessions in the cloud master instance as apposed to each individual deployed server.

**Comment by russjones** [comment]
@aashley Fixed this in #1706. The fix will be in both 2.5.0 and 2.4.2.
